### PR TITLE
Sebastian embedding finetuning tutorial

### PR DESCRIPTION
**Pull request by SebChw**
1. Changed data parsing a little bit to support current way of defying splits
2. Modified dataset so it now supports text, graph and text + graph embeddings
3. Modified dataset so that we now iterate over entire dataset during one epoch

### CODE REVIEW

File: embedding_transfer_learning/run.py [hunk 1]
```diff
@@ -1,61 +1,91 @@
...
 
 
 def getLogger(run_name: str):
     return TensorBoardLoggerAdapter(name=run_name)
 
+
 def main():
     project_name = "embedding_transfer_learning"
-    data_module = EmbeddingDataModule(batch_size=256)
+    data_module = EmbeddingDataModule(batch_size=256, num_workers=0)
```
Comment (trebacz626) [inline embedding_transfer_learning/run.py]: I now it needs to be on windows but it hurts

File: embedding_transfer_learning/run.py [hunk 2]
```diff
@@ -1,61 +1,91 @@
...
     project = ArtProject(project_name, data_module)
-    project.add_step(DataPreparation(),[])
-    project.add_step(TextDataAnalysis(),[
-        CheckResultExists("embedding_size"),
-        CheckResultExists("number_of_reviews"),
-        CheckResultExists("average_review_length"),
-        CheckResultExists("average_query_length"),
-    ])
+    project.add_step(DataPreparation(), [])
+    # project.add_step(
```
Comment (trebacz626) [inline embedding_transfer_learning/run.py]: czm zakomentowane :(

File: embedding_transfer_learning/steps.py [hunk 3]
```diff
@@ -28,86 +28,86 @@ def do(self, previous_states: Dict):
         Args:
             previous_states (Dict): previous states
         """
-        art_logger.info("Reading data")
+        if QUERIES_PATH.exists() and RECIPIES_PATH.exists():
```
Comment (trebacz626) [inline embedding_transfer_learning/steps.py:48]: Actually the hash checking was taking care of that, why is it here?

Comment (SebChw) [reply embedding_transfer_learning/steps.py:48]: If someone downloads data one doesn't have to go through preprocessing step. This step may not even run or take very long time if run on CPU

File: embedding_transfer_learning/run.py (hunk 1 above)
Comment (SebChw) [reply embedding_transfer_learning/run.py]: Ill add a parser

File: embedding_transfer_learning/run.py (hunk 2 above)
Comment (SebChw) [reply embedding_transfer_learning/run.py]: dataset nie zwraca teraz tych calych tekstow. Zmienie to w sumie

Comment (SebChw) [reply embedding_transfer_learning/run.py]: A no i w sumie, Teraz iterujemy sie po wszystkich query wiec trzeba tez bedzie ciut zmienic

**Pull request merged**